### DISCUSSION TITLE
Event handler help wanted

### DISCUSSION THREAD

**Discussion by mattholy** [fastapi/fastapi#3990] [Questions] [answered]
When use '@app.on_event()' on a multiple-worker server, I find that this handler will be called many times.I wonder if there is any thing be called only once at a server shutdown?

**Comment by mattholy** [comment]
hi anyone?

**Comment by ghandic** [comment, marked as answer]
Hey, sorry we're still transitioning from issues to discussions! You are correct, since your process manager (uvicorn or gunicorn) will run your app in isolation there are no interprocess communications. You could however have something with an external lock (most basic would be a file that you can attain a lock on and then only the worker that got the lock can run the functionality)

What is your use-case? 

If this helped solve your issue or point you in the right direction could you mark it as answered?
